rspec-core/lib/rspec/core/configuration.rb

```diff
@@ -1765,14 +1766,57 @@ def self.delegate_to_ordering_manager(*methods)
 
       # Set Ruby warnings on or off.
       def warnings=(value)
-        $VERBOSE = !!value
+        @warnings_set = true
+
+        case value
+        when true, false
+          RSpec.deprecate(
+            "RSpec::Core::Configuration#warnings=",
+            :message =>  "config.warnings = #{value} is deprecated, please set warnings to one of: " \
+            "`:none`, `:all` or `:deprecations_only.`"
```

Comment:
```suggestion
            :message =>  "config.warnings = #{value} is deprecated, please set warnings to one of: " \
                         "`:none`, `:all` or `:deprecations_only.`"
```

RuboCop suggests aligning the indentation.